cmds/dhclient/dhclient.go

```diff
@@ -174,16 +164,33 @@ func main() {
 		log.Fatalf("We're sorry, the random number generator is not up. Please file a ticket")
 	}
 
-	iList := []string{defaultIface}
+	iList := []string{defaultIfname}
 	if len(flag.Args()) > 0 {
 		iList = flag.Args()
 	}
 
+	timeout := time.Duration(*leasetimeout) * time.Second
+	// if timeout is < slop, it's too short.
+	if timeout < slop {
+		timeout = 2 * slop
+		log.Printf("increased lease timeout to %s", timeout)
+	}
+
 	done := make(chan error)
-	for _, iface := range iList {
-		go func(iface string) {
-			done <- dhclient(iface, *renewals, time.Duration(*leasetimeout)*time.Second)
-		}(iface)
+	for _, i := range iList {
+		go func(ifname string) {
+			iface, err := ifup(ifname)
```

Comment:
no. depending on the network card, it can take seconds to minutes, especially if it is NOT connected. You want to do this in the goroutine. You want to make network discovery of each card independent and conconcurrent with all cards.